### PR TITLE
add hackerspace type

### PR DESCRIPTION
**Pull request by RedAuburn**
Icon:
image
fixes https://github.com/organicmaps/organicmaps/issues/6518

### CODE REVIEW

**Comment by pastk** [comment]
> 

I get the idea, but its associated much more with some industry, mechanic or a mill..

Maybe a PCB-like pattern or a microchip? E.g.

image

image

File: iphone/Maps/LocalizedStrings/es-MX.lproj/Localizable.strings [hunk 1]
```diff
@@ -2002,15 +2002,15 @@
 /* These translations are used for all type.highway.*.bridge. */
 "type.highway.footway.bridge" = "Puente";
 
-"type.highway.footway.demanding_alpine_hiking" = "Camino";
+"type.highway.footway.demanding_alpine_hiking" = "Path";
```
Comment (pastk) [inline iphone/Maps/LocalizedStrings/es-MX.lproj/Localizable.strings:2005]: its happening again :/

File: data/categories.txt [hunk 2]
```diff
@@ -7793,6 +7793,47 @@ sw:Uwanja wa gofu|michezo
 fa:زمین گلف
 mr:गोल्फचे मैदान
 
+leisure-hackerspace
+en:4Hackerspace
+ar:هاكرزبيس
+be:Хакерская прастора
+bg:Hackerspace
+ca:Hackerspace
+cs:Hackerspace
```
Comment (pastk) [inline data/categories.txt:7802]: if there is no translation then its better to delete such a row;
`en:` version is used for search always AFAIK

File: data/editor.config [hunk 3]
```diff
@@ -1024,6 +1024,9 @@
       <type id="shop-camera" group="shop">
         <include group="poi_internet" />
       </type>
+      <type id="leisure-hackerspace" group="shop">
```
Comment (pastk) [inline data/editor.config:1027]: I don't think it belongs to shops..

File: data/strings/types_strings.txt [hunk 4]
```diff
@@ -15931,6 +15931,47 @@
...
     zh-Hant = 高爾夫球場
 
+  [type.leisure.hackerspace]
+    en = Hackerspace
+    ar = هاكرزبيس
+    be = Хакерская прастора
+    bg = Hackerspace
+    ca = Hackerspace
+    cs = Hackerspace
+    da = Hackerspace
```
Comment (pastk) [inline data/strings/types_strings.txt:15941]: better let use the EN version instead of duplicating the string

**Review by pastk — CHANGES_REQUESTED** [review]

**Comment by patepelo** [comment]
I had already been working in some icon design for hackerspace, this is what I had:

hacker_space-m
hacker_space-m (2)
hacker_space-m (3)

image

**Comment by RedAuburn** [comment]
> I had already been working in some icon design for hackerspace, this is what I had:

ah nice! i like the top left, but which one do you think we should use?

**Comment by pastk** [comment]
TBH the top left doesn't ring a bell for me, I see no association...

I like the idea of a "virtual" cube, but it stops looking like a cube when in small size :/

**Comment by patepelo** [comment]
Yeah, hackerspaces are a bit abstract and multidisciplinary, so it's hard to represent. Your electronics proposal are hard to put in a tiny size. We may use something inspired in the USB icon. 

image

Or the atom icon?

image


Or even a rocket icon

image

**Comment by patepelo** [comment]
image

**Comment by pastk** [comment]
The light bulb looks alright, I hope it won't be confused with a shop :)

A microchip
image

https://fontawesome.com/icons/microchip?f=classic&s=solid&sz=2xs&pc=%23000000

**Review by pastk — APPROVED** [review]

**Comment by biodranik** [comment]
A light bulb means a shop that sells lights.

**Comment by biodranik** [comment]
There are some interesting icons here: https://www.google.com/search?q=hackerspace+icon

Stylized H and S letters may also be ok. Colors can also be used if necessary.

**Comment by Misalf-git** [comment]
What about a "terminal console" or "command prompt"-like icon?
`>_`

When I think of "hacking" I think of _mashing on the keyboard_.

I have seen `</>` occasionally to represent any `code` (HTML, as it (I guess) resembles a programming language for a broader range of people than FORTRAN does) and got the idea immediately.

A cube reminds me of a kind of packaging service / post office / parcel locker (i.e. confusable with existing icons).

Light bulb and rocket is just something else. :)

PCB pattern, microchip and USB logo resemble techy things, but I might as well think I can bring there my hardware for repair.

**Comment by biodranik** [comment]
Recognizable command prompt is a good idea to try.

**Comment by pastk** [comment]
I guess a command prompt will be recognizable by devs and power users mostly :) But could be alright.

Wiki's (shortened) definition for a reference
> A hackerspace is a community-operated workspace where people with common interests, such as computers, machining, technology, science, digital art, or electronic art, can meet, socialize, and collaborate.

**Comment by patepelo** [comment]
> They click it and learn.

Exactly, most of the icons here are valid.

 We should find a common way to improve new icon design validation. It's hard to convince all and we end up with a pile of stalled discussions and many POIs without icon representations.

**Comment by biodranik** [comment]
My point is simple:
1. Icons that may be associated with some other shops/types are not good.
2. Icons that can be recognized at least by the target audience but not by the general audience are better than icons that can not be recognized by anyone.

**Comment by patepelo** [comment]
> My point is simple:
> 
> 1. Icons that may be associated with some other shops/types are not good.
> 2. Icons that can be recognized at least by the target audience but not by the general audience are better than icons that can not be recognized by anyone.

Your point is valid and I agree with it. 

I'm opening a broader topic that it is hard for new icons to find a clear path for design review. I'm not sure how it could be improved but we have a bottle neck. 

Maybe we could have a polling system or a design committee. I'll move this topic into it's own discussion for keeping it off the hacker-space thread.

**Comment by pastk** [comment]
I like the first one :)

**Comment by RedAuburn** [comment]
> 


This is good :) I'll make an SVG of it

**Comment by RedAuburn** [comment]
image

now using this icon 👍️ ptal @pastk

**Review by pastk — APPROVED** [review]

**Comment by patepelo** [comment]
I like that it also loos like a robot.

Thanks @Misalf-git @RedAuburn

**Comment by biodranik** [comment]
How are they looking on the map in light/dark styles?

**Comment by RedAuburn** [comment]
> How are they looking on the map in light/dark styles?

image1

**Comment by biodranik** [comment]
Thanks! IMO it looks a bit like a bug (insect) :), and too detailed/complex inside the circle when it's small (check with 18x18 pixels size).

Maybe leaving >_ is cleaner (although it may be associated with anime).

**Comment by biodranik** [comment]
Need to see it on the map in a smaller size...

**Comment by RedAuburn** [comment]
> Need to see it on the map in a smaller size...

as you wish :)

signal-2023-11-24-152907_002

**Comment by biodranik** [comment]
Definitely better without traces. Maybe also try a thinner (out)line and a more monitor-like rectangle?

**Comment by dvdmrtnz** [comment]
> with no traces:
> 

I like this one

File: data/categories.txt [hunk 5]
```diff
@@ -7793,6 +7793,22 @@ sw:Uwanja wa gofu|michezo
...
+fa:فضای هکرها
+he:האקרספייס
+ja:ハッカースペース
+ko:해커스 스페이스
+mr:हॅकरस्पेस
+ru:Хакерспейс
+th:แฮกเกอร์สเปซ
+vi:Không gian tin tặc
+zh-Hans:创客空间
+zh-Hant:駭客空間
```
[the remark this answers is not in the document]
Comment (pastk) [reply data/categories.txt:7810]: Most langs ended up with the "Hackerspace" translation :)
https://github.com/organicmaps/organicmaps/pull/6630#discussion_r1401018675

**Comment by RedAuburn** [comment]
fixed comments, icon now using thinner lines:

image

**Review by vng — APPROVED** [review]

**Pull request merged**